src/enzo/CallInSitulibyt.C

```diff
@@ -0,0 +1,452 @@
+     *  them.
+     *
+     * */
+
+    field_list[libyt_field_i].field_name = "Temperature";
+    field_list[libyt_field_i].field_type = "cell-centered";
+    field_list[libyt_field_i].field_dtype = EYT_BFLOAT;
+    for (j = 0; j < 2 * params->dimensionality; j++) {
+        field_list[libyt_field_i].field_ghost_cell[j] = NumberOfGhostZones;
+    }
+    libyt_field_i = libyt_field_i + 1;
+
+    field_list[libyt_field_i].field_name = "Cooling_Time";
+    field_list[libyt_field_i].field_type = "cell-centered";
+    field_list[libyt_field_i].field_unit = "code_time";
+    field_list[libyt_field_i].field_dtype = EYT_BFLOAT;
+    for (j = 0; j < 2 * params->dimensionality; j++) {
+        field_list[libyt_field_i].field_ghost_cell[j] = NumberOfGhostZones;
+    }
+
+    /* We now have to do everything we do in CallPython.C, which amounts to
+     *
+     *  - ExposeGridHierarchy (not necessary anymore)
+     *  - ExposeDataHierarchy (a recursive call)
+     *
+     * */
+
+    /* As I'm writing this in 2023, I cannot recall specifically why we need to
+     * find the last entry in the linked list.  But, it shows up a number of
+     * places, including in CallPython.C and OutputFromEvolveLevel.C.  I
+     * suspect it's because the grids are added in the reverse order to the
+     * LevelArray.
+     *
+     * I have left the original 'ugh:' so that it can be found by anyone
+     * grepping for that.  I don't actually feel 'ugh' about it.  In fact,
+     * in revisiting Enzo, I have found a number of places where in yt we
+     * took 'the easy way' and the Enzo way is so much more elegant, even if
+     * the impedance mismatch means I have to jump through a few more hoops.
+     *
+     * */
+
+    yt_grid* GridInfoArray;
+    yt_get_GridsPtr(&GridInfoArray);
+
+    /* These are 1-indexed, so when we access elements we -1 them */
+    int GlobalGridID = 1;
+    int LocalGridID = 1;
+    Temp2 = LevelArray[0];
+    while (Temp2->NextGridThisLevel != NULL)
+        Temp2 = Temp2->NextGridThisLevel; /* ugh: find last in linked list */
+    if(ExposeHierarchyToLibyt(MetaData, Temp2->GridHierarchyEntry, 
+                GlobalGridID, LocalGridID, CurrentTime, 0, 0, GridInfoArray) == FAIL) {
+        fprintf(stderr, "Error in ExposeHierarchyToLibyt\n");
+        return FAIL;
+    }
+
+    /* Commit all settings to libyt. */
+    if (yt_commit() != YT_SUCCESS){
+        fprintf(stderr, "Error in libyt API yt_commit\n");
+        return FAIL;
+    }
+
+    /* Run yt_run_Function and yt_run_FunctionArguments */
+	// if (yt_run_Function("yt_inline") != YT_SUCCESS) {
+	// 	   fprintf(stderr, "Error while running yt_run_Function and call yt_inline\n");
+	// 	   return FAIL;
+	// }
+
+    // if (yt_run_FunctionArguments("yt_inline_args", 1, "\'density\'") != YT_SUCCESS) {
+ 	//     fprintf(stderr, "Error while running yt_run_FunctionArguments and call yt_inline_args\n");
+	//     return FAIL;
+    // }
```

Comment:
```suggestion
        // example for calling a Python function named "yt_inline_args" that accepts arguments
        // if (yt_run_FunctionArguments("yt_inline_args",
                                    1, // number of args expected in the Python function
                                    "\'density\'" // argument passed to yt_inline_args
                                    ) != YT_SUCCESS) {
 	//     fprintf(stderr, "Error while running yt_run_FunctionArguments and call yt_inline_args\n");
	//     return FAIL;
    // }
```

Reply:
This suggestion is malformed, but hopefully conveys my intention. Since `yt_run_FunctionArguments` is used here as an example, it should be commented to better explain what the arguments are.

Reply:
I'll add more comments to this block.